MazerPlatformer/GameWorld.cs

```diff
@@ -94,30 +95,31 @@ internal void LoadContent(int levelNumber, int? overridePlayerHealth = null, int
             _rooms = _level.GetRooms();
 
             _removeWallTimer.Start();
+            return Nothing;
 
         }
 
-        private void AddToGameObjects(Dictionary<string, GameObject> levelGameObjects)
-        {
-            foreach (var levelGameObject in levelGameObjects)
-                AddToGameObjects(levelGameObject.Key, levelGameObject.Value);
-        }
+        private Either<IFailure, Unit> AddToGameObjects(Dictionary<string, GameObject> levelGameObjects)
+            => levelGameObjects
+                .Select(levelGameObject => AddToGameObjects(levelGameObject.Key, levelGameObject.Value))
+                .AggregateFailures();
 
         /// <summary>
         /// Unload the game world, and save it
         /// </summary>
-        public void UnloadContent()
-        {
-            _unloading = true;
-            _gameObjects.Clear();
-            _level.UnLoad();
-            _unloading = false;
-            _removeWallTimer.Stop();
-        }
+        public Either<IFailure, Unit> UnloadContent()
+            => Ensure(() =>
+            {
+                _unloading = true;
+                _gameObjects.Clear();
+                _level.UnLoad(); // TODO: I/O
+                _unloading = false;
+                _removeWallTimer.Stop();
+            });
```

Comment:
All this code is an eye-sore becuse you have to wrap lines of code because you don't know if they are doing to fail or not and they are also modifying underlying state that the function has no control over